middleware.ts

```diff
@@ -0,0 +1,31 @@
+// write middleware file for nextjs
+import type { NextRequest } from "next/server"
+import { NextResponse } from "next/server"
+import initializePatronTokenAuth from "./src/server/auth"
+
+/**
+ * Middleware to check and parse the nyplIdentiyPatron cookie and then verify
+ * it through JWT. The decoded patron information is then set in Nextjs'
+ * request headers.
+ */
+export async function middleware(request: NextRequest) {
+  const requestHeaders = new Headers(request.headers)
+  const patronTokenResponse = await initializePatronTokenAuth(request)
+
+  requestHeaders.set("x-is-token-valid", `${patronTokenResponse?.isTokenValid}`)
+  requestHeaders.set("x-user-id", `${patronTokenResponse?.decodedPatron?.sub}`)
```

Comment:
I was trying to not call this function on every `getServerSideProps` function/page but it seems we might need to. Not that this is bad, but I figured it could be done once in the middleware. I'll try again to make it part of a the local req or extend it instead of the custom headers.